openshift-kube-apiserver/openshiftkubeapiserver/patch.go

```diff
@@ -108,8 +108,8 @@ func OpenShiftKubeAPIServerConfigPatch(genericConfig *genericapiserver.Config, k
 	}
 	// END HANDLER CHAIN
 
-	openshiftAPIServiceReachabilityCheck := newOpenshiftAPIServiceReachabilityCheck()
-	oauthAPIServiceReachabilityCheck := newOAuthPIServiceReachabilityCheck()
+	openshiftAPIServiceReachabilityCheck := newOpenshiftAPIServiceReachabilityCheck(genericConfig.PublicAddress)
```

Comment:
this looks good, the endpoint controller reads the IP from the same field https://github.com/openshift/kubernetes/blob/6386eb2dafe47c8f56160b2829e6dc26a7ae21b8/pkg/controlplane/controller.go#L126